### PR TITLE
Require Faraday v2

### PR DESCRIPTION
**Pull request by nipe0324**
Latest Faraday version is v2. But, qiita-rb supports only v0.x.
I tried to change that qiita-rb requires Faraday v2.

### CODE REVIEW

File: spec/qiita/client_spec.rb [hunk 1]
```diff
@@ -341,6 +341,10 @@ def stub_api_request
       204
     end
 
+    let(:response_hash) do
+      nil
+    end
```
Comment (nipe0324) [inline spec/qiita/client_spec.rb:346]: Returns empty body response if 204 status code.

https://github.com/bblimke/webmock/pull/190

And, Qiita `PUT` and `DELETE` APIs return a empty body.

JSON Schema(en): https://qiita.com/api/v2/schema?locale=en

**Comment by getty104** [comment]
@nipe0324 
Thank you for creating the PR! 
The changes between Faraday v1 and v2 are significant.
So, we want to support only v2 for maintenance.
Could you please update the implementation to be compatible only with v2?

**Comment by nipe0324** [comment]
@getty104 
Thank you for your review.
I updated that qiita-rb  supports only v2.

**Comment by getty104** [comment]
@nipe0324 
Thank you for fix PR!
I'll check it.

**Review by getty104 — APPROVED** [review]
LGTM!
Thank you for your contribution!

**Pull request merged**